js/src/forum/components/DiscussionList.js

```diff
@@ -205,14 +205,4 @@ export default class DiscussionList extends Component {
       this.discussions.splice(index, 1);
     }
   }
-
-  /**
-   * Add a discussion to the top of the list.
-   *
-   * @param {Discussion} discussion
-   * @public
-   */
-  addDiscussion(discussion) {
```

Comment:
This method is used / extended by the Pusher extension, so it cannot be removed.

More importantly, we need to check whether the logic in flarum/pusher is still needed when we change this behavior here.

Reply:
The way I understand it, the hack in flarum/pusher can probably be removed: when we reload the entire discussion list anyway, any items that Pusher had in its store will be displayed anyway.

Reply:
I see, I hadn't looked at that extension, this must be the relevant code:
https://github.com/flarum/pusher/blob/00cfaa982778129f3e97e2f4e96c79d86fa4f6d9/js/src/forum/index.js#L90-L106

Unfortunately since my attempts at enabling the extension have all failed and thus I cannot test changes with it, I am going to close this PR and leave the issue for someone else.